### ISSUE TITLE
bug(web): invalid subkey triggers

### ISSUE THREAD

**Issue by jahorton** [keymanapp/keyman#10640]
While investigating something else earlier today, I found a very easy input pattern that can trigger an error we've been seeing in a few Sentry reports.

1. Load up the standard "Test unminified Keymanweb" test page while on (or emulating) a mobile device.  (Chrome emulation is fine.)
2. Using the default keyboard...
3. Press any non-special key.  (For example, the `k` key.)
4. Drag up while keeping the key pressed.

Results:
- the key-preview disappears immediately
- when the touch is released, key highlighting "sticks" and is not cleared until the key with stuck highlighting is pressed again

**Comment by jahorton** [comment]
To continue from the last @sentry-io comment:

Note the relative position of these two code blocks:

https://github.com/keymanapp/keyman/blob/e8949831da5b71cdebcc09330a57d82779890272/web/src/engine/osk/src/input/gestures/browser/subkeyPopup.ts#L87-L90

https://github.com/keymanapp/keyman/blob/e8949831da5b71cdebcc09330a57d82779890272/web/src/engine/osk/src/input/gestures/browser/subkeyPopup.ts#L146

The _latter_ of these is what causes the _first_ linked Sentry event.  In such a case, the `constructor` call fails.  But... we've _already_ attached an event-listener in an earlier block - and so, it _still has access_ to the partially-constructed object.

It's only partially constructed, though:

https://github.com/keymanapp/keyman/blob/e8949831da5b71cdebcc09330a57d82779890272/web/src/engine/osk/src/input/gestures/browser/subkeyPopup.ts#L174-L176

The block above never gets to run when the base error occurs.  So, if the gesture sequence is cancelled, triggering the `complete` event, we obtain ourselves a partially-constructed instance with `this.shim === undefined`.

**Comment by jahorton** [comment]
We have observed triggers of this since #10641's merge.  I'm unable to reproduce it in app/browser Web, but it didn't take long in Android Studio's emulation to get a trigger.

Using a well-placed conditional breakpoint, I was able to find this:

![image](https://github.com/keymanapp/keyman/assets/25213402/be2ee8cb-6466-4189-9e27-472dfc9a2f70)

After drilling into the implications of that, I think I've realized what's going on.  `validateItem` isn't being run against the KeyElement to be associated with the gesture (here, the originally-touched key) - it's always being run against the _final_ KeyElement (which may have moved).  Infrastructure needs to be added to ensure that `validateItem` is run against the correct element here.
